HousingSearchApi/V1/Helper/PostCodeHelpers.cs

```diff
@@ -28,10 +28,12 @@ public static bool SearchTextIsValidPostCode(string searchText)
 
             if (trimmed.Length > 7) return false;
 
-            const string pattern = @"^[A-Z]{1,2}[0-9]{1,2} ?[0-9][A-Z]{2}$";
+            const string pattern = @"^(([a-z][0-9]{1,2})|([a-z][a-hj-y][0-9]{1,2})|([a-z][0-9][a-z])|([a-z][a-hj-y][0-9]?[a-z]))[0-9][a-z]{2}$";
```

Comment:
> "The final two letters do not use C, I, K, M, O or V, so as not to resemble digits or each other when hand-written."

So it would probably not be [a-z]
But then again, maybe it's too overkill to validate in such detail for our usecase.